src/include/miopen/fold/problem_description.hpp

```diff
@@ -0,0 +1,481 @@
+
+struct NetworkConfig;
+
+namespace fold {
+
+bool checkSameLength(const TensorDescriptor& x, const TensorDescriptor& y);
+
+struct UnfoldFwdProblemDescription : ProblemDescriptionBase
+{
+    UnfoldFwdProblemDescription(const TensorDescriptor& inputDesc_,
+                                const TensorDescriptor& outputDesc_,
+                                const int32_t* kernel_size_,
+                                const int kernel_size_size_,
+                                const int32_t* stride_,
+                                const int stride_size_,
+                                const int32_t* padding_,
+                                const int padding_size_,
+                                const int32_t* dilation_,
+                                const int dilation_size_)
+        : inputDesc(inputDesc_),
+          outputDesc(outputDesc_),
+          kernel_size(kernel_size_),
+          kernel_size_size(kernel_size_size_),
+          stride(stride_),
+          stride_size(stride_size_),
+          padding(padding_),
+          padding_size(padding_size_),
+          dilation(dilation_),
+          dilation_size(dilation_size_)
+    {
+        IsValidSize();
+        IsValidType();
+    }
+
+    bool IsValidSize() const
+    {
+        if(inputDesc.GetSize() != 4)
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Unfold: The input tensor should be 4D.");
+#else
+            return false;
+#endif
+        }
+        int spatial_dim_size = inputDesc.GetSize() - 2;
+        if(kernel_size_size != spatial_dim_size || stride_size != spatial_dim_size ||
+           padding_size != spatial_dim_size || dilation_size != spatial_dim_size)
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Unfold: Argument length should be 2D");
+#else
+            return false;
+#endif
+        }
+        auto input_dims = inputDesc.GetLengths();
+        const int32_t N = static_cast<int32_t>(input_dims[0]);
+        const int32_t C = static_cast<int32_t>(input_dims[1]);
+        int32_t P = 1, L = 1;
+        std::vector<int32_t> ls;
+        for(int i = 0; i < spatial_dim_size; ++i)
+        {
+            P *= kernel_size[i];
+            int32_t l = (static_cast<int32_t>(input_dims[i + 2]) + 2 * padding[i] -
+                         dilation[i] * (kernel_size[i] - 1) - 1) /
+                            stride[i] +
+                        1;
+            L *= l;
+            ls.push_back(l);
+        }
+        std::vector<size_t> output_dims_desired{
+            static_cast<size_t>(N), static_cast<size_t>(C * P), static_cast<size_t>(L)};
+        auto output_dims = outputDesc.GetLengths();
+        if(output_dims != output_dims_desired)
+        {
+#if MIOPEN_BUILD_DEV || !MIOPEN_NDEBUG
+            MIOPEN_THROW(miopenStatusBadParm, "Unfold: Invalid output dimension");
+#else
+            return false;
+#endif
+        }
+        return true;
```

Comment:
I think it's okay to delete this part as it is not used anywhere.

Reply:
resolve in commit [8ee2861](https://github.com/moreh-dev/MIOpen/pull/37/commits/8ee286160f08a51be318531311032182c8cb3a12)